metadata/reportssql/testing_report_functions.sql

```diff
@@ -553,6 +577,85 @@ BEGIN
 END$$ 
 DELIMITER ;
 
+-- getPatientAgeInMonthsAtDate
+
+DROP FUNCTION IF EXISTS getPatientAgeInMonthsAtDate;
+
+DELIMITER $$
+CREATE FUNCTION getPatientAgeInMonthsAtDate(
+    p_patientId INT(11),
+    p_date DATE) RETURNS VARCHAR(50)
+    DETERMINISTIC
+BEGIN
+    DECLARE result VARCHAR(50);
+
+    SELECT 
+        timestampdiff(MONTH, p.birthdate, p_date) INTO result 
+    FROM person p 
+    WHERE p.voided = 0
+        AND p.person_id = p_patientId
+    LIMIT 1;
+
+    RETURN result;
+END$$
+DELIMITER ;
+
+-- getTestResultWithinReportingPeriod
+
+DROP FUNCTION IF EXISTS getTestResultWithinReportingPeriod;
+
+DELIMITER $$
+CREATE FUNCTION getTestResultWithinReportingPeriod(
+    p_patientId INT(11),
+    p_startDate DATE,
+    p_endDate DATE,
+    p_testUuid VARCHAR(38),
+    p_testDateUuid VARCHAR(38)) RETURNS VARCHAR(50)
+    DETERMINISTIC
+BEGIN
+    DECLARE result VARCHAR(50);
+
+    -- retrieve the test result from OpenElis
+    SELECT cn.name INTO result
+    FROM obs o
+        JOIN concept_name cn ON o.value_coded = cn.concept_id AND cn.locale = "en"
+    WHERE o.voided = 0
+        AND o.person_id = p_patientId
+        AND o.order_id IS NOT NULL
+        AND o.concept_id = (SELECT c.concept_id FROM concept c WHERE c.uuid = p_testUuid LIMIT 1)
+        AND o.date_created BETWEEN p_startDate AND p_endDate
```

Comment:
@tmvumbi2 Test case 12 failed. We should probably add a order by date desc here? 

https://docs.google.com/spreadsheets/d/1drmjK2JUgSeAOraH_jDIuE-kqMsNwN-lhZ1Q6QkWGGw/edit#gid=726254203

Reply:
This works, thanks!